presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -186,15 +196,22 @@ public void close()
             return;
         }
         closed = true;
-        probe = null;
-        pageBuilder.reset();
+        reachedProbeEnd = true;
+        reset();
         // closing lookup source is only here for index join
         if (lookupSource != null) {
             lookupSource.close();
         }
         onClose.run();
     }
 
+    private boolean readyToBuildOutputPage()
+    {
+        // Because of dictionary, always output when probe reached the end. Problem is indexes will refer
```

Comment:
`s/Problem is indexes will refer to different probe pages./Indexes cannot span multiple probe pages/`